client/components/SideNav/SideNav.tsx

```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+import styles from "./SideNav.module.css";
+import Logo from "./../../assets/logo.svg";
+
+export default function SideNav() {
+  const [search, setSearch] = useState("");
+
+  const handleSearch = (value: string) => {
+    setSearch(value);
+  };
```

Comment:
So instead of doing the search logic in here, do it outside of SideNav() so create a new function 
```suggestion
const SearchInput = ({ onSearch }) => {
  return (
    <div className={styles["input-icons"]}>
      <i className="fa fa-search icon"></i>
      <input
        type="text"
        placeholder="Search for..."
        onChange={(e) => onSearch(e.target.value)}
      />
    </div>
  );
};
```

Reply:
Also outside create another function called NavButton to be able to reuse this style of button
```
const NavButton = ({ children, onClick }) => {
  return (
    <button onClick={onClick}>
      {children}
    </button>
  );
};
```

Reply:
Ack!